src/main/kotlin/graphics/scenery/backends/vulkan/VU.kt

```diff
@@ -38,561 +39,487 @@ fun BigInteger.toHexString(): String {
+    fun VkDevice.newCommandBuffer(commandPool: VkCommandPool, level: VkCommandBufferLevel = VkCommandBufferLevel.PRIMARY, autostart: Boolean = false): VkCommandBuffer {
+        val cmdBuf = getCommandBuffer(commandPool, level)
 
-                VkCommandBuffer(commandBuffer, device.vulkanDevice)
-            }
-        }
+        if (autostart)
+            cmdBuf.begin()
 
-        fun newCommandBuffer(device: VulkanDevice, commandPool: Long, level: Int = VK_COMMAND_BUFFER_LEVEL_PRIMARY, autostart: Boolean = false): VkCommandBuffer {
-            val cmdBuf = newCommandBuffer(device, commandPool, level)
+        return cmdBuf
+    }
 
-            if(autostart) {
-               beginCommandBuffer(cmdBuf)
-            }
+    fun newCommandBuffer(device: VulkanDevice, commandPool: VkCommandPool, level: Int = VK_COMMAND_BUFFER_LEVEL_PRIMARY, autostart: Boolean = false): VkCommandBuffer {
+        val cmdBuf = device.vulkanDevice.getCommandBuffer(commandPool, VkCommandBufferLevel of level)
 
-            return cmdBuf
-        }
+        if (autostart)
+            cmdBuf.begin()
 
-        fun beginCommandBuffer(commandBuffer: VkCommandBuffer, flags: Int = VK_COMMAND_BUFFER_USAGE_SIMULTANEOUS_USE_BIT) {
-            stackPush().use { stack ->
-                val cmdBufInfo = VkCommandBufferBeginInfo.callocStack(stack)
-                    .sType(VK_STRUCTURE_TYPE_COMMAND_BUFFER_BEGIN_INFO)
-                    .pNext(NULL)
-                    .flags(flags)
+        return cmdBuf
+    }
 
-                VU.run("Beginning command buffer", { vkBeginCommandBuffer(commandBuffer, cmdBufInfo) }, {})
-            }
+    fun beginCommandBuffer(commandBuffer: VkCommandBuffer, flags: Int = VK_COMMAND_BUFFER_USAGE_SIMULTANEOUS_USE_BIT) {
+        stackPush().use { stack ->
+            val cmdBufInfo = VkCommandBufferBeginInfo.callocStack(stack)
+                .sType(VK_STRUCTURE_TYPE_COMMAND_BUFFER_BEGIN_INFO)
+                .pNext(NULL)
+                .flags(flags)
+
+            VU.run("Beginning command buffer", { vkBeginCommandBuffer(commandBuffer, cmdBufInfo) }, {})
         }
+    }
 
-        fun createDescriptorSetLayout(device: VulkanDevice, type: Int = VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER_DYNAMIC, binding: Int = 0, descriptorNum: Int = 1, descriptorCount: Int = 1, shaderStages: Int = VK_SHADER_STAGE_ALL): Long {
-            return stackPush().use { stack ->
-                val layoutBinding = VkDescriptorSetLayoutBinding.callocStack(descriptorNum, stack)
-                (binding until descriptorNum).forEach { i ->
-                    layoutBinding[i]
-                        .binding(i)
-                        .descriptorType(type)
-                        .descriptorCount(descriptorCount)
-                        .stageFlags(shaderStages)
-                        .pImmutableSamplers(null)
-                }
+    fun createDescriptorSetLayout(
```

Comment:
Indentation looks like one level too little, could that be?

Reply:
no idea, I'll double auto format the code